### ISSUE TITLE
Use of var in amf-client-js/amf.js

### ISSUE THREAD

**Issue by jarrodek** [aml-org/amf#888]
The generated `amf.js` file contains two `require` declarations as the following:

```javascript
SHACLValidator = require("amf-shacl-node");
Ajv = require("ajv");
// rest of the code
```
This works for a regular scripts but it fails as an ES module which requires defining variables the correct way. Please, add `var` before variables declarations so the library can be used as a ES module:

```javascript
var SHACLValidator = require("amf-shacl-node");
var Ajv = require("ajv");
```

**Comment by tomsfernandez** [comment]
Hi @jarrodek. Can you elaborate a bit more on why this change is required and why do you need this change? (I know you said that you want to use it as an ES module but I'm not grasping the meaning of it)

**Comment by jarrodek** [comment]
Hi @tomsfernandez ,
The ES modules can only run in strict mode and there's nothing you can do about it. I have a requirement to run my application in a m ES module. Therefore everything imported as an ES module (via both `import` and `import()`) are in strict mode. This causes an error because both variables are not defined at the moment of using them and therefore the whole script is crashing.

Thanks

**Comment by tomsfernandez** [comment]
@jarrodek I have created an internal issue to track this. The issue is: https://www.mulesoft.org/jira/browse/APIMF-3020. All updates/questions will be in this Github thread.

**Comment by arielmirra** [comment]
Hi @jarrodek! Quick question, does this count as valid js in strict mode:
```javascript
var global = global;
global.SHACLValidator = require("amf-shacl-node");
global.Ajv = require("ajv");
```

Simply adding `var` to the beginning makes tests fail with errors like `Cannot find global Ajv object`. You can try this solution by editing the `amf.js` file. Let me know if it works and I'll continue testing from there. Thanks!

**Comment by arielmirra** [comment]
Hi @jarrodek! is this issue still relevant to you?

**Comment by jarrodek** [comment]
Sorry for not responding. GitHub does not notify me about mentions for some reason. 
Global is not the best way to solve this. There is no such variable in a web environment (it is in Node  but not in browser). Instead you can use [globalThis](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis) which refers to the global object across different environments.

P.S. It's faster to Slack me then waiting for the answer here.

**Comment by arielmirra** [comment]
Hi @jarrodek! We've fixed the issue in [APIMF-3020](https://www.mulesoft.org/jira/browse/APIMF-3020) and it'll be available in the release of AMF 4.7.4. I'm closing the issue.
